### DISCUSSION TITLE
Why nuget?

### DISCUSSION THREAD

**Discussion by MOZGIII** [Squirrel/Squirrel.Windows#1770] [Q&A] [answered]
As I understand, NuGet itself is intended for sipping libraries to developers, not apps to the end-users. Why does Squirrel.Windows uses NuGet?

I mean, there's definitely a learning curve to get a hold of NuGet, while we don't really get to use the NuGet's hosted package repos or anything like that. There are also some semantic limitations of the package format itself.

I see how the patch-update functionality of NuGet is tempting - but in that it?

I just doubt whether NuGet makes sense in the long run - i.e. if there's enough time to implement patch-updates functionality and tooling in a more portable way (i.e. such that it works for all Squirrel platforms, not just Windows) - it doesn't make sense to involve NuGet at all. Am I wrong here?

**Comment by anaisbetts** [comment, marked as answer]
> I just doubt whether NuGet makes sense in the long run

Squirrel has used NuGet for nearly a decade, this wasn't exactly a recent decision 😅 

Indeed, were I to write Squirrel today, given its userbase, I would probably not choose NuGet packages as its starting point - Squirrel was originally designed for C# apps, where having accessible tooling (i.e. NuGet Package Explorer where you can drag-and-drop stuff) and an automated toolset that built packages (msbuild packaging) was compelling. 

But breaking changes in large ecosystems are Extremely Expensive, so until there is an Extremely Compelling reason to change it (with the developer time budget to match), it will stay as-is
